component/component.go

```diff
@@ -107,3 +107,25 @@ type ConfigUnmarshaler interface {
 // intoCfg interface{}
 //   An empty interface wrapping a pointer to the config struct to unmarshal into.
 type CustomUnmarshaler func(componentViperSection *viper.Viper, intoCfg interface{}) error
+
+// CreateParamsBase is passed to Create* functions.
+type CreateParamsBase struct {
+	// ApplicationStartInfo can be used by components for informational purposes
+	ApplicationStartInfo ApplicationStartInfo
+}
+
+// ApplicationStartInfo is the information that is logged at the application start and
+// passed into each component. This information can be overridden in custom builds.
+type ApplicationStartInfo struct {
```

Comment:
If we are passing this info into each component, I thought it makes sense to move the struct here. Technically this is a breaking change, but very minor